Set-DomainUser.ps1

```diff
@@ -1,720 +1,1118 @@
+    $NETBIOS = Get-DomainNetbiosName
+}
+If ($DomainNameAutodetected) {
+    Write-Host "Detected the following domain names. Is this correct?"
+    Write-Host "DNS name:     $domain"
+    Write-Host "NETBIOS name: $NETBIOS"
+    $DomainConfirmPrompt = Read-Host "Please type 'y' for yes or 'n' for no."
+    if ($DomainConfirmPrompt -ne 'y') {
+        Write-Host "Please rerun the script and provide the correct domain DNS and NETBIOS names on the command line."
+        exit 1
+    }
+}
+# Test PSM credentials
+if ($TestPsmConnectCredentials) {
+    if (ValidateCredentials -domain $domain -Credential $psmConnectCredentials) {
+        Write-Host "PSMConnect user credentials validated"
+    }
+    else {
+        Write-Error "PSMConnect user validation failed. Please validate PSMConnect user name and password or remove -TestPsmConnectCredentials to skip this test"
+        exit 1
+    }
+}
+    
+if ($TestPsmAdminCredentials) {
+    if (ValidateCredentials -domain $domain -Credential $psmAdminCredentials) {
+        Write-Host "PSMAdminConnect user credentials validated"
+    }
+    else {
+        Write-Error "PSMAdminConnect user validation failed. Please validate PSMConnect user name and password or remove -TestPsmAdminCredentials to skip this test."
+        exit 1
+    }
+}
+    
+# Reverse logic on script invocation setting because double negatives suck
+$DoHardening = !$DoNotHarden
+$DoConfigureAppLocker = !$DoNotConfigureAppLocker
+    
+Write-Host "Logging in to CyberArk"
+$pvwaToken = New-ConnectionToRestAPI -pvwaAddress $pvwaAddress -tinaCreds $tinaCreds
+if (Test-PvwaToken -Token $pvwaToken -pvwaAddress $pvwaAddress) {
+    Write-Host "Successfully logged in"
+}
+else {
+    Write-Host "Error logging in to CyberArk"
+    exit 1
+}
+# Get platform info
+Write-Host "Checking current platform status"
+$platformStatus = Get-PlatformStatus -pvwaAddress $pvwaAddress -pvwaToken $pvwaToken -PlatformId $PlatformName
+if ($platformStatus -eq $false) {
+    # function returns false if platform does not exist
+    # Creating Platform
+    Write-Host "Creating new platform"
+    Duplicate-Platform -pvwaAddress $pvwaAddress -pvwaToken $pvwaToken -CurrentPlatformId "7" -NewPlatformName $PlatformName -NewPlatformDescription "Platform for PSM accounts"
+    $Tasks += ("Set appropriate policies and settings on platform `"{0}`"" -f $PlatformName)
+    # Get platform info again so we can ensure it's activated
+    $platformStatus = Get-PlatformStatus -pvwaAddress $pvwaAddress -pvwaToken $pvwaToken -PlatformId $PlatformName
+}
+else {
+    Write-Warning ('Platform {0} already exists. Please verify it meets requirements.' -f $PlatformName)
+    $Tasks += ("Verify that the existing platform `"{0}`" is configured correctly" -f $PlatformName)
+}
+if ($platformStatus.Active -eq $false) {
+    Write-Host "Platform is deactivated. Activating."
+    Activate-Platform -pvwaAddress $pvwaAddress -pvwaToken $pvwaToken -Platform $platformStatus.Id
+}
```

Comment:
- Check platform status
- Create if it doesn't exist and check it again
- Activate if it's deactivated